router/customdestinationmanager/customdestinationmanager.go

```diff
@@ -317,7 +329,9 @@ func (customManager *CustomManagerT) backendConfigSubscriber() {
 				}
 			}
 		}
-		customManager.mapLock.Unlock()
+		once.Do(func() {
+			customManager.backendConfigInitialized <- struct{}{}
+		})
```

Comment:
To me now this method looks like "it's fine if you call it multiple times but we'll tell you when the intialization is done only the first time".
I wonder if the whole initilization should be "onced" then and not just the last bit? wdyt?

Reply:
it is not about calling the method more than one, it is about reading from the channel more than once, so a new `sync.Once` would be required for what you describe, correct?

Reply:
I was under the impression that it doesn't make sense to call `customManager.backendConfigSubscriber()` more than once because once subscribed I guess it'll keep getting configs as they come from the backend config. So I was proposing to have the entire content of that method inside `once.Do()` or, if possible, it would be even better to remove the `sync.Once` altogether because it seems to me that we're invoking this method only once per `customManager` instance inside the `New()` constructor [here](https://github.com/rudderlabs/rudder-server/blob/fca1c1d12fd7a026e442c83e7914354f6b8a4ae6/router/customdestinationmanager/customdestinationmanager.go#L304).

But perhaps I'm missing something, let me know if I'm not making sense.